include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -102,19 +104,47 @@ class Cube : public std::enable_shared_from_this<Cube> {
 
     /// Is the current cube root.
     [[nodiscard]] bool is_root() const noexcept;
+    /// Is the current cube a leaf.
+    [[nodiscard]] bool is_leaf() const noexcept;
     /// At which child level this cube is.
     /// root cube = 0
     [[nodiscard]] std::size_t grid_level() const noexcept;
     /// Count the number of Type::SOLID and Type::NORMAL cubes.
     [[nodiscard]] std::size_t count_geometry_cubes() const noexcept;
 
+    [[nodiscard]] glm::vec3 center() const noexcept {
+        return m_position + 0.5f * m_size;
+    }
+
+    [[nodiscard]] const glm::vec3 &position() const noexcept {
+        return m_position;
+    }
+
+    [[nodiscard]] std::array<glm::vec3, 2> bounding_box() const {
+        return {m_position, {m_position.x + m_size, m_position.y + m_size, m_position.z + m_size}};
+    }
+
+    [[nodiscard]] float bounding_box_radius() const {
```

Comment:
Does this refer to the bounding sphere? If yes better `bounding_sphere_radius`

Reply:
That's right. It's actually the radius of the bounding sphere!

Reply:
Done.